test/common/stats/utility_test.cc

```diff
@@ -273,6 +275,15 @@ TEST_P(StatsUtilityTest, ScopeTextReadoutOnce) { scopeOnce<TextReadout>(makeText
 
 TEST_P(StatsUtilityTest, ScopeTextReadoutAll) { scopeAll<TextReadout>(makeTextReadout()); }
 
+TEST_P(StatsUtilityTest, SanitizeStatsName) {
+  EXPECT_EQ("a.b.c", Utility::sanitizeStatsName("a.b.c."));
+  EXPECT_EQ("a.b.c", Utility::sanitizeStatsName(".a.b.c"));
+  EXPECT_EQ("a__b", Utility::sanitizeStatsName("a::b"));
+  EXPECT_EQ("a", Utility::sanitizeStatsName("a.\0"));
```

Comment:
these are great. I think though that "a" is not the result you intended, but it should work if you do:
```
EXPECT_EQ("a._", Utility::sanitizeStatsName(absl::string_view("a.\0", 3)));
```

Reply:
Done.